homeassistant/components/weather/__init__.py

```diff
@@ -126,14 +129,19 @@ def temperature(self) -> float | None:
 
     @property
     def temperature_unit(self) -> str:
-        """Return the unit of measurement."""
+        """Return the unit of measurement for temperature."""
```

Comment:
Let's make it clear in the docstring this is the weather entity's unit, before conversion, same for the other units.